migtests/scripts/functions.sh

```diff
@@ -599,6 +602,175 @@ import_data_status(){
+        "target")
+            expected_count=$(jq -r '.target_delta_events // empty' "$metadata_file" 2>/dev/null)
+            ;;
+    esac
+    
+    # Validate the count is a positive number
+    if [[ "$expected_count" =~ ^[1-9][0-9]*$ ]]; then
+        echo "$expected_count"
+    fi
+}
+
+# Helper function to count actual events for a specific exporter database using data-migration-report
+# Usage: count_exported_events <exporter_db>
+# Returns: number of events found
+count_exported_events() {
+    local exporter_db="$1"
+    local total_count=0
+    
+    # Validate exporter_db parameter
+    case "$exporter_db" in
+        "source"|"target")
+            # Valid exporter_db
+            ;;
+        *)
+            echo "0"
+            return
+            ;;
+    esac
+    
+    # Run the command to generate the data migration report file (suppress all output)
+    get_data_migration_report >/dev/null 2>&1
+    local report_file="${EXPORT_DIR}/reports/data-migration-report.json"
+    
+    if [ -f "$report_file" ] && [ -s "$report_file" ]; then
+        # Validate JSON and count total exported events (inserts + updates + deletes) for the specified exporter_db
+        if jq empty "$report_file" 2>/dev/null; then
+            total_count=$(jq -r --arg exporter_db "$exporter_db" '
+                if type == "array" then
+                    map(select(.db_type == $exporter_db)) | 
+                    map(.exported_inserts + .exported_updates + .exported_deletes) | 
+                    add // 0
+                else
+                    0
+                end
+            ' "$report_file" 2>/dev/null || echo "0")
+        else
+            # Invalid JSON, return 0
+            total_count=0
+        fi
+    else
+        # Report file doesn't exist or is empty, return 0
+        total_count=0
+    fi
+    echo "$total_count"
+}
+
+# Wait until sufficient events from a specific exporter database are detected
+# which ensures that the exporter is capturing changes and cutover can be initiated safely
+# Usage: wait_for_exporter_event <exporter_db> [timeout_seconds]
+# Supported exporter_db values:
+#   - source (for source database change events)
+#   - target (for target database change events in fall-forward/fall-back scenarios)
+wait_for_exporter_event() {
+    local exporter_db="$1"
+    local timeout_seconds="${2:-300}"
+
+    if [ -z "$exporter_db" ]; then
+        echo "wait_for_exporter_event: exporter_db is required"
+        return 1
+    fi
+
+    # Try to get expected count from metadata file
+    local expected_count=$(get_expected_event_count "$exporter_db")
+    
+    local step_message="Wait for events from exporter_db=${exporter_db}"
+    if [ -n "$expected_count" ]; then
+        step_message="${step_message} (expecting ${expected_count} events)"
+    else
+        step_message="${step_message} (fallback: first event + 10s)"
+    fi
+    step "$step_message"
+
+    # Ensure we're in streaming mode (quick wait)
```

Comment:
Shall we be increasing this wait for checking if streaming phase has started?
Or do we need this check? remove it?
@makalaaneesh @shubham-yb

Reply:
this check probably isn't useful, you're right. 
Ultimately we are checking the get-data-migration-report after this, so we will anyway have to wait for that. So, why additionally wait for export_status STREAMING.